### PR TITLE
Updated error message in case wrong variant is inpput.

### PR DESCRIPTION
**Pull request by namsonx**

### CODE REVIEW

File: RobotFramework_TestsuitesManagement/Config/CConfig.py [hunk 1]
```diff
@@ -520,17 +520,25 @@ def __loadConfigFileLevel2(self):
             CConfig.sLoadedCfgError = str(error)
             logger.error(f"Loading of JSON configuration file failed! Reason: {CConfig.sLoadedCfgError}")
             raise Exception
-        
+        if self.sConfigName not in oSuiteConfig:
+            CConfig.sLoadedCfgError = f"Testsuite management - Loading configuration level 2 failed! \n \
+        The variant '{self.sConfigName}' is not defined in '{os.path.abspath(self.sTestSuiteCfg)}'"
```
Comment (test-fullautomation) [inline RobotFramework_TestsuitesManagement/Config/CConfig.py]: Hi Son,
in TML I display here a list of all possible variants.
It should be possible to get this from the json file which defines the variants.
Thank you,
 Thomas

**Review by test-fullautomation — APPROVED** [review]
Hi Son,
thank you for your pull request.
Please read my findings.

Thank you,
 Thomas

File: RobotFramework_TestsuitesManagement/Config/CConfig.py (hunk 1 above)
Comment (namsonx) [reply RobotFramework_TestsuitesManagement/Config/CConfig.py]: Hello Thomas,

I added the information of possible variants in the second commit of this pull request.

Thank you,
Son

File: RobotFramework_TestsuitesManagement/Config/CConfig.py [hunk 2]
```diff
@@ -522,7 +522,8 @@ def __loadConfigFileLevel2(self):
             raise Exception
         if self.sConfigName not in oSuiteConfig:
             CConfig.sLoadedCfgError = f"Testsuite management - Loading configuration level 2 failed! \n \
```
Comment (test-fullautomation) [inline RobotFramework_TestsuitesManagement/Config/CConfig.py:527]: Hi Son,
please confirm the proper look, or if a concatenation would be a better solution for the line-break.
Thank you,
 Thomas

**Review by test-fullautomation — APPROVED** [review]
Hi Son,
please look into my finding.
Thank you,
 Thomas

**Review by test-fullautomation — CHANGES_REQUESTED** [review]
.

File: RobotFramework_TestsuitesManagement/Config/CConfig.py [hunk 3]
```diff
@@ -522,7 +522,8 @@ def __loadConfigFileLevel2(self):
...
-        The variant '{self.sConfigName}' is not defined in '{os.path.abspath(self.sTestSuiteCfg)}'"
+        The variant '{self.sConfigName}' is not defined in '{os.path.abspath(self.sTestSuiteCfg)}' \n \
+        Please find the suitable variant in this list: {list(oSuiteConfig.keys())}"
```
Comment (test-fullautomation) [inline RobotFramework_TestsuitesManagement/Config/CConfig.py]: Hi Son,
very good!
small findings:
- single quotes per entry are missing
- if the variant names are many and long its difficult to read, therefore please align properly with other content of the log file and log possible variants line by line.

Pleaes check how TML behaves as reference.

Just trigger a BVT with a wrong variant name.

Thank you,
 Thomas

**Review by test-fullautomation — CHANGES_REQUESTED** [review]
Hi Son,
please check my findings.
Thank you,
 Thomas

File: RobotFramework_TestsuitesManagement/Config/CConfig.py (hunk 2 above)
Comment (namsonx) [reply RobotFramework_TestsuitesManagement/Config/CConfig.py:527]: Hi Thomas,

I have just updated log output of variants list. Now the log look like below:
[ ERROR ] Testsuite management - Loading configuration level 2 failed! 
         The variant 'variant_23' is not defined in 'C:\Users\mas2hc\Desktop\data_storage\workspace\ROBFWAIO-OSS\robotframework-testsuitesmanagement\atest\general_config\testsuites_config.json'
         Please find the suitable variant in this list:
            - 'default'
            - 'variant_1'
            - 'variant_2'

Thank you,
Son

**Review by HolQue — CHANGES_REQUESTED** [review]
Hi Son,

error message looks really fine now!

Next step should be to check also if the variant name itself is a proper one.

Currently: "`The variant './localconfig/exercise-pg_localconfig_bench1.json' is not defined in ...`"

It makes no sense to check, if this variant is defined because this cannot be the name of a variant. Before you check the definition you should check if the name itself is valid.

See also end of

https://github.com/test-fullautomation/robotframework-testsuitesmanagement/issues/95#issuecomment-1411604797

Variant names in command line (variant) should be checked against characters that are not allowed inside names (like path separators). This rule would detect errors like
`--variable variant:"D:/path/to/configfile.json"`
where accidently a path to a config file is provided instead of a variant name.

**Review by HolQue — CHANGES_REQUESTED** [review]
Hi Son,

please check the entire code for similar lines.

When I provide a variant containing blanks only:

`--variable variant:"    "`

then I get:

`The variant '' is not defined in ...`

Better would be:

`The variant name must not be empty ...`

And the check of the variant name against not allowed characters is still open.

**Review by test-fullautomation — APPROVED** [review]
Hi Son,
looks good to me now!
Thank you,
 Thomas

**Pull request merged**